cli/mender-artifact/read.go

```diff
@@ -84,18 +87,78 @@ func readArtifact(c *cli.Context) error {
 	fmt.Printf("  Version: %d\n", info.Version)
 	fmt.Printf("  Signature: %s\n", sigInfo)
 	fmt.Printf("  Compatible devices: '%s'\n", r.GetCompatibleDevices())
+	provides := ar.GetArtifactProvides()
+	if provides != nil {
+		fmt.Printf("  Provides group: %s\n", provides.ArtifactGroup)
+	}
+
+	depends := ar.GetArtifactDepends()
+	if depends != nil {
+		fmt.Printf("  Depends on one of artifact(s): [%s]\n", strings.Join(depends.ArtifactName, ", "))
+		fmt.Printf("  Depends on one of group(s): [%s]\n", strings.Join(depends.ArtifactGroup, ", "))
+	}
+
 	if len(scripts) > -1 {
 		fmt.Printf("  State scripts:\n")
 	}
 	for _, scr := range scripts {
 		fmt.Printf("    %s\n", scr)
 	}
-	fmt.Printf("\nUpdates:\n")
 
+	fmt.Printf("\nUpdates:\n")
 	for k, p := range inst {
 		fmt.Printf("  %3d:\n", k)
-		fmt.Printf("    Type:   %s\n", p.GetType())
-		for _, f := range p.GetUpdateFiles() {
+		fmt.Printf("    Type:   %s\n", p.GetUpdateType())
+
+		provides, err := p.GetUpdateProvides()
+		fmt.Printf("    Provides:")
+		if err != nil {
+			fmt.Printf(" Invalid provides section: %s\n", err.Error())
+		} else if provides == nil || len(*provides) == 0 {
+			fmt.Printf(" Nothing\n")
+		} else {
+			fmt.Printf("\n")
+			for provideKey, provideValue := range *provides {
+				fmt.Printf("\t%s: %s\n", provideKey, provideValue)
+			}
+		}
+
+		depends, err := p.GetUpdateDepends()
+		fmt.Printf("    Depends:")
+		if err != nil {
+			fmt.Printf(" Invalid depends section: %s\n", err.Error())
+		} else if depends == nil || len(*depends) == 0 {
+			fmt.Printf(" Nothing\n")
+		} else {
+			fmt.Printf("\n")
+			for dependKey, dependValue := range *depends {
+				fmt.Printf("\t%s: %s\n", dependKey, dependValue)
+			}
+		}
+
+		metaData, err := p.GetUpdateMetaData()
+		fmt.Printf("    Metadata:")
+		if err != nil {
+			fmt.Printf(" Invalid metadata section: %s\n", err.Error())
+		} else if len(metaData) == 0 {
+			fmt.Printf(" Nothing\n")
+		} else {
+			var metaDataSlice []byte
+			if err == nil {
+				metaDataSlice, err = json.Marshal(metaData)
+			}
+			var metaDataBuf bytes.Buffer
+			if err == nil {
+				err = json.Indent(&metaDataBuf, metaDataSlice, "\t", "  ")
+			}
+			if err != nil {
+				fmt.Printf(" Invalid metadata section: %s\n", err.Error())
+			} else {
+				fmt.Printf("\n\t%s\n", metaDataBuf.String())
+			}
+		}
+
+		for _, f := range p.GetUpdateAllFiles() {
```

Comment:
I only spent a limited time on the printing feature, since I assume most of it will be superseded or influenced by #155.